### ISSUE TITLE
compiler warnings

### ISSUE THREAD

**Issue by rettinghaus** [rism-digital/verovio#1653]
@craigsapp please fix these warnings:
```
verovio/src/iohumdrum.cpp:17113:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
verovio/src/iohumdrum.cpp:17114:29: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
verovio/src/iohumdrum.cpp:17113:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
verovio/src/iohumdrum.cpp:8022:21: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Space *>' requested here
verovio/src/iohumdrum.cpp:17113:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
verovio/src/iohumdrum.cpp:15338:15: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Chord *>' requested here
verovio/src/iohumdrum.cpp:17113:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
verovio/src/iohumdrum.cpp:15639:5: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Rest *>' requested here
verovio/src/iohumdrum.cpp:17113:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
verovio/src/iohumdrum.cpp:16262:19: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Note *>' requested here
verovio/src/hum/humlib.cpp:36305:12: Implicit conversion loses integer precision: 'long' to 'int'
verovio/src/hum/humlib.cpp:36362:12: Implicit conversion loses integer precision: 'long' to 'int'
verovio/src/hum/humlib.cpp:36429:12: Implicit conversion loses integer precision: 'long' to 'int'
verovio/src/hum/humlib.cpp:36902:12: Implicit conversion loses integer precision: 'long' to 'int'
verovio/src/hum/humlib.cpp:36951:12: Implicit conversion loses integer precision: 'long' to 'int'
verovio/src/hum/humlib.cpp:37128:12: Implicit conversion loses integer precision: 'long' to 'int'
verovio/src/hum/humlib.cpp:37517:12: Implicit conversion loses integer precision: 'long' to 'int'
verovio/src/hum/humlib.cpp:38620:15: Implicit conversion loses integer precision: 'long' to 'int'
verovio/src/hum/humlib.cpp:39792:12: Implicit conversion loses integer precision: 'size_t' (aka 'unsigned long') to 'int'
verovio/src/hum/humlib.cpp:72037:28: Implicit conversion loses integer precision: 'unsigned long' to 'int'
verovio/src/hum/humlib.cpp:72919:59: Implicit conversion loses integer precision: 'std::__1::vector<hum::GridVoice *, std::__1::allocator<hum::GridVoice *> >::size_type' (aka 'unsigned long') to 'int'

verovio/src/iohumdrum.cpp:17518:13: Variable 'ntrack' may be uninitialized when used here
verovio/src/iohumdrum.cpp:17513:15: Initialize the variable 'ntrack' to silence this warning
verovio/src/iohumdrum.cpp:17550:22: Variable 'ntrack' may be uninitialized when used here
verovio/src/iohumdrum.cpp:17545:15: Initialize the variable 'ntrack' to silence this warning
verovio/src/hum/humlib.cpp:47860:7: Variable 'beat1' may be uninitialized when used here
verovio/src/hum/humlib.cpp:47835:11: Initialize the variable 'beat1' to silence this warning
```

**Comment by valeriyvan** [comment]
Nice catch, @rettinghaus! These warnings are so annoying! So nice to see project clean from warnings and issues.

> @craigsapp please fix these warnings:
>

**Comment by craigsapp** [comment]
If you turn off the warnings you will not see them 🙀

**Comment by craigsapp** [comment]
Related to this issue, I will warn @lpugin about this commit: https://github.com/rism-ch/verovio/commit/ae6b4950f7e64a891c98bfa7e8fdc714dcd147e0, where I allow `<staff>` to be directly inserted into `<section>` rather than using `<measure>` as an intermediary layer.  I will be merging `develop-humdrum` into `develop` to fix this current issue, and that change will come with the updates from that commit as well.

This change is to allow for mensural music to be created inside of verovio.  I have not encountered any problems related to that change, but there is probably a more formal method of allowing it, and/or have some sort of mensural mode.   Also consideration should be given to allowing mensural and CMN notation as different staves on the same system.

**Comment by craigsapp** [comment]
There are two compiler warnings for the current version of the `develop` branch:

```
[  7%] Building CXX object CMakeFiles/verovio/src/beam.cpp.o
/verovio/src/beam.cpp:299:39: warning: comparison of integers of different
      signs: 'std::__1::vector<vrv::BeamElementCoord *, std::__1::allocator<vrv::BeamElementCoord
      *> >::size_type' (aka 'unsigned long') and 'int' [-Wsign-compare]
    if (m_beamElementCoordRefs.size() != nbRests) {
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~
```

and

```
/verovio/src/turn.cpp:29:7: warning: base class 'vrv::AttOrnamentAccid' will
      be initialized after base 'vrv::AttExtSym' [-Wreorder]
    , AttOrnamentAccid()
      ^
```

These are from compiling on MacOS (clang version 11.0.3).  I note there are no compiler warnings for `iohumdrum.cpp` or `humlib.cpp`; because otherwise I am as OCD as you two about compiler warnings and remove them when I see them.  MacOS compiling has fewer compiler warnings than on Linux with GCC and Microsoft Visual Studio.  I usually check Linux/GCC v10 before merging into develop, but rarely check MS compiler warnings until they are reported to me :-).

Travis compiling with LLVM 9.0.0 does not have any warnings for humlib: https://travis-ci.org/github/craigsapp/humlib/jobs/726633207

Travis compiling on Linux with GCC 5 does not have any warnings for humlib: https://travis-ci.org/github/craigsapp/humlib/jobs/726633206, but it looks like something is not configured right since it does not look like anything compiled.  In any case I compile locally with GCC 10 to check for warnings from GCC.

Appveyor on Visual studio 2017 is reporting warnings which I will fix: https://ci.appveyor.com/project/craigsapp/humlib/build/job/i3sgw4n78h1jx7eh
These are probably the source of the warnings listed above.  Although @rettinghaus usually compiles on Linux?  Where I am not seeing the warnings.

Here was an attempt to get more warnings on MacOS: https://github.com/rism-ch/verovio/issues/1107

**Comment by craigsapp** [comment]
No warnings in Travis tests:
Xcode: xcode11.1 C++/AMD64: https://travis-ci.org/github/rism-ch/verovio/jobs/726699611
linux-g++/AMD64: https://travis-ci.org/github/rism-ch/verovio/jobs/726699612

However for linux-g++/AMD64 there is a new warning in note.cpp:

```
/home/travis/build/rism-ch/verovio/src/note.cpp:580:37: warning: enumeral and non-enumeral type in conditional expression [-Wextra]
         = (GetDur() >= DURATION_16) ? SMUFL_E242_flag16thUp : flag->GetSmuflCode(GetDrawingStemDir());
```

No warnings for:
 Precache emscripten/AMD64 https://travis-ci.org/github/rism-ch/verovio/jobs/726699613

No warnings for:
Emscripten default (with humdrum)/AMD64: https://travis-ci.org/github/rism-ch/verovio/jobs/726699617

No warnings for:
Appveyor Visual Studio 2017/AMD64: https://ci.appveyor.com/project/LaurentPugin/verovio-r1t6l

There is a new warning not mentioned on other compilers:

```
 iomusxml.cpp
C:\projects\verovio-r1t6l\src\iomusxml.cpp(3692): warning C4566: character represented by universal-character-name '\u20E5' cannot be represented in the current code page (1252) [C:\projects\verovio-r1t6l\tools\verovio.vcxproj]
C:\projects\verovio-r1t6l\src\iomusxml.cpp(3693): warning C4566: character represented by universal-character-name '\u0338' cannot be represented in the current code page (1252) [C:\projects\verovio-r1t6l\tools\verovio.vcxproj]
```

**Comment by craigsapp** [comment]
@rettinghaus: can you recompile the current `develop` branch and post humdrum-related warnings?

I did not fix most of the ones in the original post (or at least do not remember fixing them since the last merge):

```
verovio/src/iohumdrum.cpp:17113:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
verovio/src/iohumdrum.cpp:17114:29: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
verovio/src/iohumdrum.cpp:17113:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
etc.
```

So maybe your compiler is more picky than the ones I have tested it on.  The line numbers should now match in both `develop` and `develop-humdrum` branches, so I can work on removing any remaining warnings from your list.


Most of those are complaints of long to int, which should indicate Unix/Linux OSes.  I did not get them on Red Hat Linux for some reason, but you are probably using Ubuntu to compile?

https://en.cppreference.com/w/cpp/language/types

<img width="717" alt="Screen Shot 2020-09-12 at 10 12 06 PM" src="https://user-images.githubusercontent.com/3487289/93010805-26c1b900-f545-11ea-8437-6ea0fd28dd31.png">

The LP64 model has a minimum of 32 bits for int, and 64 for long.  Maybe on your linux OS, the int is 32-bit, but on my linux OS it is 64 bits, so there are no complaints about a loss of precision for me.

One reason why I use int, is that I learned it was the "fast" integer type (at least in 16- and 32-bit architectures), and matched the word size of the computer architecture.  That is probably not true in 64-bits unix/linux since the integer is not the word size for that architecture.  I see that in C++11 there are special data types for selecting the fastest data type with any minimum given size:

https://en.cppreference.com/w/cpp/types/integer

<img width="739" alt="Screen Shot 2020-09-12 at 10 21 52 PM" src="https://user-images.githubusercontent.com/3487289/93010921-5e7d3080-f546-11ea-80b5-8d4d80d5e549.png">

So when I use `int`, I am intending `int_fast32_t` which is new for C++11.  This value would be 64-bits if the 64-bit int is faster for the CPU to handle, but 32-bit if the 32-bit int is the fastest to process.   Probably `int` is not faster than `long` on unix/linux 64-bit architectures since it would have to be unpacked from the 64-bit words coming from memory before the CPU can do arithmetic on it.

`intptr_t` is an interesting C++11 type.  I should be using that in `HumInstrument.cpp` where I have the search currently implemented in C and using pointer arithmetic.

**Comment by rettinghaus** [comment]
@craigsapp Actually I can't remember exactly where I stumbled upon these warnings. What I can say is, that it was the latest develop version on a 64-bit POSIX system, so definitively not 32-bit. 

Haven't seen any other warnings now except the ones you've reported.

**Comment by craigsapp** [comment]
>  on a 64-bit POSIX system, so definitively not 32-bit.

Yes, but I was mentioning that on 64-bit linux computers, the `int` is usually 32 bits (not suggesting a 32-bit architecture), and that could be the source of the warnings.  On a 32-bit linux computer, the `int` and the `long` should be the same size, so those warnings should not occur.

**Comment by rettinghaus** [comment]
@craigsapp here are some more for you to work on. 

I figured out (again) were they came from: trying to build `Verovio Framework` in Xcode.

```
/iohumdrum.cpp:17537:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
/iohumdrum.cpp:17538:29: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
/iohumdrum.cpp:17537:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
/iohumdrum.cpp:8309:21: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Space *>' requested here
/iohumdrum.cpp:17537:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
/iohumdrum.cpp:15728:15: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Chord *>' requested here
/iohumdrum.cpp:17537:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
/iohumdrum.cpp:16046:5: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Rest *>' requested here
/iohumdrum.cpp:17537:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
/iohumdrum.cpp:16673:19: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Note *>' requested here
/hum/humlib.cpp:36305:12: Implicit conversion loses integer precision: 'long' to 'int'
/hum/humlib.cpp:36362:12: Implicit conversion loses integer precision: 'long' to 'int'
/hum/humlib.cpp:36429:12: Implicit conversion loses integer precision: 'long' to 'int'
/hum/humlib.cpp:36902:12: Implicit conversion loses integer precision: 'long' to 'int'
/hum/humlib.cpp:36951:12: Implicit conversion loses integer precision: 'long' to 'int'
/hum/humlib.cpp:37128:12: Implicit conversion loses integer precision: 'long' to 'int'
/hum/humlib.cpp:37517:12: Implicit conversion loses integer precision: 'long' to 'int'
/hum/humlib.cpp:38620:15: Implicit conversion loses integer precision: 'long' to 'int'
/hum/humlib.cpp:39792:12: Implicit conversion loses integer precision: 'size_t' (aka 'unsigned long') to 'int'
/iohumdrum.cpp:17942:13: Variable 'ntrack' may be uninitialized when used here
/iohumdrum.cpp:17937:15: Initialize the variable 'ntrack' to silence this warning
/iohumdrum.cpp:17974:22: Variable 'ntrack' may be uninitialized when used here
/iohumdrum.cpp:17969:15: Initialize the variable 'ntrack' to silence this warning
/hum/humlib.cpp:47860:7: Variable 'beat1' may be uninitialized when used here
/hum/humlib.cpp:47835:11: Initialize the variable 'beat1' to silence this warning
```

**Comment by craigsapp** [comment]
OK, I will get to work on them (although it is late here, so not immediately).

It would be handy to figure out if there is a warning flag for these from XCode that will also work in the command-line tools, since they do not post any of those errors in MacOS... (maybe `-Wall` if it is not already being used?)

----------------------------

Here are the type sizes in MacOS 64-bit, with ints being 4 bytes and longs being 8 bytes:

<img width="278" alt="Screen Shot 2020-09-17 at 1 38 32 AM" src="https://user-images.githubusercontent.com/3487289/93446918-a376cf00-f886-11ea-8207-86607eb0eb31.png">

https://stackoverflow.com/questions/2383520/on-mac-os-x-in-c-on-a-64-bit-cpu-is-there-a-type-that-is-64-bits

**Comment by musicEnfanthen** [comment]
(@lpugin , @craigsapp , @rettinghaus)

There are still some compiler warnings in the gh-actions output depending on the build system (for evnironment specs of gh-actions see: https://github.com/actions/virtual-environments): 

**On ubuntu 20-04 with gcc-9**

1. https://github.com/rism-ch/verovio/runs/1269293497?check_suite_focus=true#step:6:129
```
 [ 49%] Building CXX object CMakeFiles/verovio.dir/home/runner/work/verovio/verovio/src/object.cpp.o
/home/runner/work/verovio/verovio/src/note.cpp: In member function ‘bool vrv::Note::IsDotOverlappingWithFlag(vrv::Doc*, int, bool)’:
/home/runner/work/verovio/verovio/src/note.cpp:580:43: warning: enumeral and non-enumeral type in conditional expression [-Wextra]
  580 |         = (this->GetDur() >= DURATION_16) ? SMUFL_E242_flag16thUp : flag->GetSmuflCode(GetDrawingStemDir());
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 49%] Building CXX object CMakeFiles/verovio.dir/home/runner/work/verovio/verovio/src/octave.cpp.o
```

2. https://github.com/rism-ch/verovio/runs/1269293497?check_suite_focus=true#step:6:139
```
[ 52%] Building CXX object CMakeFiles/verovio.dir/home/runner/work/verovio/verovio/src/pb.cpp.o
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-dollar-in-identifier-extension’
[ 52%] Building CXX object CMakeFiles/verovio.dir/home/runner/work/verovio/verovio/src/pedal.cpp.o
```

3. https://github.com/rism-ch/verovio/runs/1269293497?check_suite_focus=true#step:6:190
```
[ 78%] Building CXX object CMakeFiles/verovio.dir/home/runner/work/verovio/verovio/src/tuplet.cpp.o
/home/runner/work/verovio/verovio/src/options.cpp: In constructor ‘vrv::Options::Options()’:
/home/runner/work/verovio/verovio/src/options.cpp:607:1: note: variable tracking size limit exceeded with ‘-fvar-tracking-assignments’, retrying without
  607 | Options::Options()
      | ^~~~~~~
[ 78%] Building CXX object CMakeFiles/verovio.dir/home/runner/work/verovio/verovio/src/turn.cpp.o
```

[3] not thrown with **Ubuntu gcc-10**


**On ubuntu 20-04 with clang 6**

[1]-[3] not thrown, but:

4. https://github.com/rism-ch/verovio/runs/1269293518?check_suite_focus=true#step:6:216
```
[ 94%] Building CXX object CMakeFiles/verovio.dir/home/runner/work/verovio/verovio/libmei/atts_gestural.cpp.o
/home/runner/work/verovio/verovio/src/hum/humlib.cpp:78152:48: warning: using floating point absolute value function 'fabs' when argument is of integer type [-Wabsolute-value]
                        } else if (doubleQ && (lastbarnum > -1) && (fabs(barnum - lastbarnum) > 1)) {
                                                                    ^
/home/runner/work/verovio/verovio/src/hum/humlib.cpp:78152:48: note: use function 'std::abs' instead
                        } else if (doubleQ && (lastbarnum > -1) && (fabs(barnum - lastbarnum) > 1)) {
                                                                    ^~~~
                                                                    std::abs
[ 95%] Building CXX object CMakeFiles/verovio.dir/home/runner/work/verovio/verovio/libmei/atts_externalsymbols.cpp.o
```

[4] also thrown with **clang 9 and 10**


**On MacOs with xcode default (clang latest)**
runs without warnings

**On MacOs with gcc**
runs without warnings

**Comment by rettinghaus** [comment]
@musicEnfanthen thanks. I'll investigate.

**Comment by rettinghaus** [comment]
@lpugin @musicEnfanthen 
1. is obviously correct code and misinterpreted by GCC. We could turn this warning off, but probably this isn't a good idea, because there are many places in Verovio where we could end in and enumeral and non-enumeral comparison. 
2. comes from a compiler option, that GCC doesn't understand like Clang does. Perhaps you can check this @musicEnfanthen?
3. not so much we can do in Options according to https://stackoverflow.com/questions/23499909/adjust-variable-tracking-assignment-length

**Comment by lpugin** [comment]
> 1. is obviously correct code and misinterpreted by GCC. We could turn this warning off, but probably this isn't a good idea, because there are many places in Verovio where we could end in and enumeral and non-enumeral comparison.

Maybe the warning can be removed with an additional assignment? Something like:
```
wchar_t flagGlyph = SMUFL_E242_flag16thUp;
data_DURATION dur = this->GetDur();
if (dur < DURATION_16) flagGlyph = flag->GetSmuflCode(GetDrawingStemDir());
```

**Comment by musicEnfanthen** [comment]
Perfect, after #1765 [1] and [2] seem to be fixed. 

Only [3] is showing up with gcc 9 ("not so much we can do"), and [4] with clang builds on Ubuntu.  

@craigsapp Would it be safe to use `abs` instead of `fabs` in https://github.com/rism-ch/verovio/blob/develop/src/hum/humlib.cpp#L78152-L78154

**Comment by craigsapp** [comment]
> @craigsapp Would it be safe to use abs instead of fabs in https://github.com/rism-ch/verovio/blob/develop/src/hum/humlib.cpp#L78152-L78154

Yes that will not be a problem (and it will be desireable since the input is an integer).   I changed it in humlib but have not integrated into verovio yet (I will need to check across multiple compilers to see if I have added additional warnings :-).  I will change `fabs()` to `abs()` in the `develop` branch in the meantime, and that update will be overwritten by the same change later one.

**Comment by craigsapp** [comment]
`fabs()` &rarr; `abs()` warning fixed in commit https://github.com/rism-ch/verovio/commit/96925f5927b93baa3db1b371ad12a71203e1879b.

**Comment by musicEnfanthen** [comment]
Thank you! 

Unfortunately, there seems to be an incident https://www.githubstatus.com/incidents/k46g5dn2c4lt that affects the artifact upload: https://github.com/actions/upload-artifact/issues/137

**Comment by musicEnfanthen** [comment]
Could you just try to re-run the workflow here: https://github.com/rism-ch/verovio/actions/runs/318514605 (there should be a re-run button in the upper right corner, visible to anyone with access rights)

**Comment by musicEnfanthen** [comment]
It should basically run without compiler warnings and errors: https://github.com/musicEnfanthen/verovio/runs/1283555477?check_suite_focus=true

**Comment by rettinghaus** [comment]
Following warnings are still not resolved:
```
VerovioFramework Group
Value Conversion Issue Group
/verovio/src/iohumdrum.cpp:17537:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
/verovio/src/iohumdrum.cpp:17538:29: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
/verovio/src/iohumdrum.cpp:17537:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
/verovio/src/iohumdrum.cpp:8309:21: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Space *>' requested here
/verovio/src/iohumdrum.cpp:17537:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
/verovio/src/iohumdrum.cpp:15728:15: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Chord *>' requested here
/verovio/src/iohumdrum.cpp:17537:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
/verovio/src/iohumdrum.cpp:16046:5: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Rest *>' requested here
/verovio/src/iohumdrum.cpp:17537:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *> >::difference_type' (aka 'long') to 'int'
/verovio/src/iohumdrum.cpp:16673:19: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Note *>' requested here
/verovio/src/hum/humlib.cpp:36305:12: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:36362:12: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:36429:12: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:36902:12: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:36951:12: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:37128:12: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:37517:12: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:38620:15: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:39792:12: Implicit conversion loses integer precision: 'size_t' (aka 'unsigned long') to 'int'
Semantic Issue Group
/verovio/src/iohumdrum.cpp:17942:13: Variable 'ntrack' may be uninitialized when used here
/verovio/src/iohumdrum.cpp:17937:15: Initialize the variable 'ntrack' to silence this warning
/verovio/src/iohumdrum.cpp:17974:22: Variable 'ntrack' may be uninitialized when used here
/verovio/src/iohumdrum.cpp:17969:15: Initialize the variable 'ntrack' to silence this warning
/verovio/src/hum/humlib.cpp:47860:7: Variable 'beat1' may be uninitialized when used here
/verovio/src/hum/humlib.cpp:47835:11: Initialize the variable 'beat1' to silence this warning
```

**Comment by rettinghaus** [comment]
Updated list
```
Verovio-Humdrum Group
Value Conversion Issue Group
/verovio/src/iohumdrum.cpp:19691:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *>>::difference_type' (aka 'long') to 'int'
/verovio/src/iohumdrum.cpp:19692:29: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *>>::difference_type' (aka 'long') to 'int'
/verovio/src/iohumdrum.cpp:19691:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *>>::difference_type' (aka 'long') to 'int'
/verovio/src/iohumdrum.cpp:8716:21: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Space *>' requested here
/verovio/src/iohumdrum.cpp:19691:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *>>::difference_type' (aka 'long') to 'int'
/verovio/src/iohumdrum.cpp:17558:15: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Chord *>' requested here
/verovio/src/iohumdrum.cpp:19691:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *>>::difference_type' (aka 'long') to 'int'
/verovio/src/iohumdrum.cpp:17995:5: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Rest *>' requested here
/verovio/src/iohumdrum.cpp:19691:30: Implicit conversion loses integer precision: 'typename iterator_traits<__wrap_iter<char *>>::difference_type' (aka 'long') to 'int'
/verovio/src/iohumdrum.cpp:18703:19: In instantiation of function template specialization 'vrv::HumdrumInput::convertRhythm<vrv::Note *>' requested here
/verovio/src/hum/humlib.cpp:28036:22: Implicit conversion loses integer precision: 'unsigned long' to 'int'
/verovio/src/hum/humlib.cpp:28081:22: Implicit conversion loses integer precision: 'unsigned long' to 'int'
/verovio/src/hum/humlib.cpp:36876:12: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:36933:12: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:37000:12: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:37473:12: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:37522:12: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:37699:12: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:38088:12: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:39191:15: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:40363:12: Implicit conversion loses integer precision: 'size_t' (aka 'unsigned long') to 'int'
/verovio/src/hum/humlib.cpp:48033:14: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:48034:14: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:48035:14: Implicit conversion loses integer precision: 'long' to 'int'
/verovio/src/hum/humlib.cpp:48043:11: Implicit conversion loses integer precision: 'size_t' (aka 'unsigned long') to 'int'
```

**Comment by rettinghaus** [comment]
Fixed with ae469b38b8d194d27587d0312e722c0468bb0a4f.
